### ISSUE TITLE
size mismatch for jester pre-trained model

### ISSUE THREAD

**Issue by olgapeled** [ahmetgunduz/Real-time-GesRec#11]
Hi,

I am trying to apply offline_test.py on the jester dataset with your pre-trained model.
and I got:
"size mismatch for module.conv1.weight: copying a param with shape torch.Size([64, 3, 7, 7, 7]) from checkpoint, the shape in current model is torch.Size([64, 3, 3, 7, 7])."

I think that maybe I have some problem with my parameters.
Can you please help me?

Thank you again!
I appreciate all you help!

**Comment by ahmetgunduz** [comment]
Hi olga,
please see my answer  here:
https://github.com/ahmetgunduz/Real-time-GesRec/issues/5#issuecomment-481688086

**Comment by olgapeled** [comment]
I saw this issue but it is for other dataset. For jester dataset, I use mode  RGB. I can't understand what changes I need to do? shortcut of the network should be B? 
Sorry for all my questions !

**Comment by ahmetgunduz** [comment]
> shortcut of the network should be B?
yes because ResNeXt requires `B`. 

I am here to answer. :) 

It does not check actuallly, I just wanted to remind myself if the model trained has first 3 is for RGB modality as RGB has three channels. Seconf 3 is for first item of kernel size. For example if you change it to _modify_first_conv_layer(model,3,7) everything will work because the pretrained model has kernel of (7, 7, 7)

**Comment by olgapeled** [comment]
Great!!! Thank you for your explanation.  The change should be :_modify_first_conv_layer(model,7,3).
And, for GPU, the model and the input tensor should be converted to cuda in the code. 
Last question regarding jester dataset, can you please share the accuracy (precision & recall) that you got on this dataset? 
I read the paper and I understand that you used jester dataset only as the first step in the training process, but, I am interested especially in this dataset. I have only RGB images.

**Comment by ahmetgunduz** [comment]
Thanks for the classifying the parameters and cuda setting! 

|    Model    	|   input   	| Modality (RGB) 	|
|:-----------:	|:---------:	|:--------------:	|
|     C3D     	| 16-frames 	|      82.01     	|
|     C3D     	| 24-frames 	|      87.85     	|
|     C3D     	| 32-frames 	|      94.14     	|
| ResNeXt-101 	| 32-frames 	|      95.72     	|

This is the results I got for the Jester dataset, for which I did not do aggressive  fine-tune the hyper parameters search since it is used only an initialization.  These models can be improved but I think It would not worth the effort.  I could not share the C3D model because I was training with a different setting and it would harm current project settings. But the ResNeXt-101 is the one I shared. You can test the models by simply changing the `main.py` to `offline_test.py` in run_offline.sh. 

You may see my thesis for more details. https://github.com/ahmetgunduz/TUM_Thesis

**Comment by olgapeled** [comment]
Thank you for sharing your thesis! 
With the pre-trained you published on jester "jester_resnext_101_RGB_32.pth" you got 95.72 on the validation set ??? prec@1?

**Comment by ahmetgunduz** [comment]
exactly! You may try it yourself by using `run_online.sh` by changing main.py to offline_test.py.

**Comment by olgapeled** [comment]
I did it! but, I got:

Overall Prec@1 0.63552% Prec@5 0.80280%

Any ideas why is that?

**Comment by olgapeled** [comment]
Those are my parameters:
resnet_shortcut : "B"
nesterov : false
dataset : "jester"
resume_path : "jester_resnext_101_RGB_32.pth"
mean
0 : 114.7748
1 : 107.7354
2 : 99.475
no_softmax_in_test : false
std
0 : 38.7568578
1 : 37.88248729
2 : 40.02898126
test : false
scale_step : 0.84089641525
sample_duration : 32
weighted : false
learning_rate : 0.1
checkpoint : 10
momentum : 0.9
no_train : false
lr_patience : 10
ft_begin_index : 0
video_path : "20bn-jester-v1"
n_val_samples : 3
train_crop : "corner"
n_threads : 4
manual_seed : 1
scales
0 : 1
1 : 0.84089641525
2 : 0.7071067811803005
3 : 0.5946035574934808
4 : 0.4999999999911653
mean_dataset : "activitynet"
no_cuda : false
n_finetune_classes : 27
weight_decay : 0.001
model : "resnext"
result_path : "./results"
lr_steps
0 : 10
1 : 25
2 : 50
3 : 80
4 : 100
resnext_cardinality : 32
no_val : false
begin_epoch : 1
annotation_path : "jester.json"
n_epochs : 200
scale_in_test : 1
wide_resnet_k : 2
sample_size : 112
root_path : "."
pretrain_path : ""
model_depth : 101
no_hflip : false
train_validate : false
optimizer : "sgd"
test_subset : "val"
norm_value : 1
no_mean_norm : false
initial_scale : 1
batch_size : 1
n_scales : 5
store_name : "model"
std_norm : false
crop_position_in_test : "c"
modality : "RGB"
n_classes : 27
arch : "resnext-101"
dampening : 0.9

**Comment by olgapeled** [comment]
Found my mistake!! n_val_samples MUST be set to 1!!
Thank you very much!!!

**Comment by ahmetgunduz** [comment]
that is great to hear!
